packages/wonder-blocks-core/components/clickable-behavior.js

```diff
@@ -269,13 +298,81 @@ export default class ClickableBehavior extends React.Component<
         this.dragging = false;
     }
 
+    async runCallbackAndMaybeNavigate(e: SyntheticEvent<>) {
+        const {
+            onClick = undefined,
+            beforeNav = undefined,
+            safeWithNav = undefined,
+            skipClientNav,
+            history,
+        } = this.props;
+        let shouldNavigate = true;
+
+        if (onClick) {
+            onClick(e);
+        }
+
+        // If onClick() has called e.preventDefault() then we shouldn't
+        // navigate.
+        if (e.defaultPrevented) {
+            shouldNavigate = false;
+        }
+
+        // Prevent navigation.
+        e.preventDefault();
+
+        try {
+            if (beforeNav) {
+                this.setState({waiting: true});
+                await beforeNav();
+            }
+
+            if (safeWithNav) {
+                if (history && !skipClientNav) {
```

Comment:
Shouldn't this be `history && skipClientNav`? It seems like we're trying to cover the case where we don't do client-side navigation, which I think happens when `skipClientNav` is true.